### DISCUSSION TITLE
`azd init -t [template]` to generate a folder by default, with name [template]

### DISCUSSION THREAD

**Discussion by anfibiacreativa** [Azure/azure-dev#4297] [Ideas]
Right now, when running `azd init -t [template]` , the code gets copied to the folder the devs are at. I am not sure how tools for other languages work, but both cloning a repo, or using `npm init`(create@[tool]) generate a folder with project name.

I propose the following

- In `azd` general config, ability to set a --root flag when devs want the code copied to the same directory they're executing the command on, otherwise create a folder with name == template name
- the --root can be passed as argument

**Comment by vhvb1989** [comment]
IIRC, we decided to use the current folder so you can immediately run `azd up` after `azd init`; without running `cd folder` first.

If you want to use a new folder, you can first create that folder and then use `-C ./folderName`, like

```
mkdir folder-name
azd init -t todo-nodejs-mongo -C ./folderName
```

The arg -C or -cwd is a global flag for all azd commands
![image](https://github.com/user-attachments/assets/7760ac21-7357-49c2-83e9-1024f5b9b524)

One thing we could do is ask the user if they want to create the folder if it is not found; instead of failing and expecting the folder to exists.
